Tests/FluentAssertions.Specs/Types/TypeExtensionsSpecs.cs

```diff
@@ -124,6 +125,33 @@ public void When_getting_fake_implicit_conversion_operator_from_a_type_with_fake
         result.Should().NotBeNull();
     }
 
+    [Theory]
+    [InlineData(typeof(MyRecord), true)]
+    [InlineData(typeof(MyRecordStruct), true)]
+    [InlineData(typeof(MyRecordStructWithOverriddenEquality), true)]
+    [InlineData(typeof(MyReadonlyRecordStruct), true)]
+    [InlineData(typeof(MyStruct), false)]
+    [InlineData(typeof(MyStructWithOverriddenEquality), false)]
+    [InlineData(typeof(MyClass), false)]
+    [InlineData(typeof(int), false)]
+    [InlineData(typeof(string), false)]
+    public void IsRecord_should_detect_records_correctly(Type type, bool expected)
```

Comment:
🤔 Is there a way to _not_ directly exercise tests against `IsRecord` and instead use the real API surface (`BeEquivalentTo`)?
🌱 (thinking out loud/seed) In retrospect, we should have kept `IsRecord` extension method inside the `Equivalency` folder

Reply:
I guess that really depends on personal preference and what is one's view about units and composition. In one case you would test a smaller unit which is notable its own (in this case a pure function, besides caching, which is always nice to test 😄), then give its functionality for granted in broader context. At the other end you would exercise the code indirectly, hiding what could be an implementation detail at the expenses of a more verbose test (usually in the arrange part), resulting in a more "end to end" approach, if you allow me to stretch the meaning.
My personal preference is guided from the cardinality and complexity of scenarios, and in this case it leans towards the former approach, but feel free to dispose of the donated code as you see fit!

Reply:
I think that's an acceptable trade-off.